garage/src/hooks/usePersistentState.ts

```diff
@@ -0,0 +1,54 @@
+import { useState, useEffect } from "react";
+
+// Utility functions for cookies
+const setCookie = (name: string, value: string, days: number): void => {
+  const expires = new Date(Date.now() + days * 864e5).toUTCString();
+  document.cookie = `${name}=${encodeURIComponent(
+    value
+  )}; expires=${expires}; path=/`;
+};
+
+const getCookie = (name: string): string | null => {
+  const value = `; ${document.cookie}`;
+  const parts = value.split(`; ${name}=`);
+  if (parts.length === 2)
+    return decodeURIComponent(parts.pop()!.split(";").shift()!);
+  return null;
+};
+
+export const usePersistentState = <T>(
+  key: string,
+  defaultValue: T
+): [T, React.Dispatch<React.SetStateAction<T>>] => {
+  const isLocalStorageAvailable = (() => {
+    try {
+      localStorage.setItem("test", "test");
```

Comment:
should these values be "test"? or does it not really matter what they are?

Reply:
yeah it doesn't matter what the values are, it is just testing if localStorage is available and falling back to a cookie if it isn't. also ht gpt for writing this